Issue with CMAKE_MAKE_PROGRAM not being set on Linux

Hello!

I am using `scikit-build-core` successfully in OpenCAMLib and am now also adding it to OpenVoronoi.
I noticed however, in a github actions linux (ubuntu-22.04) runner, and also on my ubuntu VM, that there is an issue with finding the ninja make program.

 It seems to find ninja correctly (`CMAKE_GENERATOR: Using ninja: /tmp/pip-build-env-w5n0jxac/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja`)

But then somehow doesn't set `CMAKE_MAKE_PROGRAM` (correctly?).

Here is a log:

```
 Installing backend dependencies ... done
  Running command Preparing metadata (pyproject.toml)
  2022-12-16 11:52:52,871 - scikit_build_core - DEBUG - RUN: /usr/bin/cmake --version
  2022-12-16 11:52:52,878 - scikit_build_core - INFO - CMake version: 3.22.1
  2022-12-16 11:52:52,887 - scikit_build_core - DEBUG - Default generator: Unix Makefiles
  2022-12-16 11:52:52,887 - scikit_build_core - DEBUG - RUN: /tmp/pip-build-env-w5n0jxac/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja --version
  2022-12-16 11:52:52,889 - scikit_build_core - INFO - Ninja version: 1.11.1
  2022-12-16 11:52:52,889 - scikit_build_core - DEBUG - CMAKE_GENERATOR: Using ninja: /tmp/pip-build-env-w5n0jxac/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja
  2022-12-16 11:52:52,889 - scikit_build_core - DEBUG - /tmp/tmpyiqown8o/build/CMakeInit.txt:
    set(SKBUILD [===[2]===] CACHE STRING "")
    set(SKBUILD_CORE_VERSION [===[0.1.4]===] CACHE STRING "")
    set(SKBUILD_PROJECT_NAME [===[openvoronoi]===] CACHE STRING "")
    set(SKBUILD_PROJECT_VERSION [===[2022.12.9]===] CACHE STRING "")
    set(PYTHON_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(PYTHON_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(PYTHON_LIBRARY [===[/usr/lib/aarch64-linux-gnu/libpython3.10.so]===] CACHE PATH "")
    set(Python_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(Python_ROOT_DIR [===[/usr]===] CACHE STRING "")
    set(Python_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(Python_FIND_REGISTRY [===[NEVER]===] CACHE STRING "")
    set(Python3_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(Python3_ROOT_DIR [===[/usr]===] CACHE STRING "")
    set(Python3_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(Python3_FIND_REGISTRY [===[NEVER]===] CACHE STRING "")
    set(SKBUILD_SOABI [===[cpython-310-aarch64-linux-gnu]===] CACHE STRING "")
    set(SKBUILD_PLATLIB_DIR [===[/tmp/tmpyiqown8o/wheel/platlib]===] CACHE PATH "")
    set(SKBUILD_DATA_DIR [===[/tmp/tmpyiqown8o/wheel/data]===] CACHE PATH "")
    set(SKBUILD_HEADERS_DIR [===[/tmp/tmpyiqown8o/wheel/headers]===] CACHE PATH "")
    set(SKBUILD_SCRIPTS_DIR [===[/tmp/tmpyiqown8o/wheel/scripts]===] CACHE PATH "")
    set(SKBUILD_NULL_DIR [===[/tmp/tmpyiqown8o/wheel/null]===] CACHE PATH "")
  2022-12-16 11:52:52,889 - scikit_build_core - DEBUG - RUN: /usr/bin/cmake -S. -B/tmp/tmpyiqown8o/build -C/tmp/tmpyiqown8o/build/CMakeInit.txt -DCMAKE_BUILD_TYPE=Release -DBUILD_PYTHON_MODULE=ON -DBoost_ADDITIONAL_VERSIONS=1.80.0;1.79.0;1.78.0;1.77.0;1.76.0;1.75.0;1.74.0;1.73.0;1.72.0;1.71.0;1.70.0 -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-w5n0jxac/overlay/local/lib/python3.10/dist-packages/scikit_build_core/resources/find_python -DCMAKE_PREFIX_PATH:PATH=/usr/local/lib/python3.10/dist-packages;/tmp/pip-build-env-w5n0jxac/overlay/local/lib/python3.10/dist-packages -D CMAKE_BUILD_TYPE=Release -D Boost_ADDITIONAL_VERSIONS=1.81.0;1.80.0;1.79.0;1.78.0;1.77.0;1.76.0;1.75.0;1.74.0;1.73.0;1.72.0;1.71.0;1.70.0
  loading initial cache file /tmp/tmpyiqown8o/build/CMakeInit.txt
  CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
  CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
  CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
  -- Configuring incomplete, errors occurred!
  See also "/tmp/tmpyiqown8o/build/CMakeFiles/CMakeOutput.log".
  *** scikit-build-core 0.1.4 using CMake 3.22.1
  *** Configurating CMake...

  *** CMake configuration failed
  error: subprocess-exited-with-error
  
  \u00d7 Preparing metadata (pyproject.toml) did not run successfully.
  \u2502 exit code: 1
  \u2570\u2500> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python3 /home/koen/.local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpl1ppdmym
  cwd: /home/koen/openvoronoi
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

\u00d7 Encountered error while generating package metadata.
\u2570\u2500> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

I currently work around this by settings `CMAKE_GENERATOR` to "Unix Makefiles" on linux, but this seems to be an issue in scikit-build-core.

Let me know if I can help debugging this, or if you need more information.

```
koen@linuxke:~/openvoronoi$ python3 -m pip install . --verbose
Using pip 22.3.1 from /home/koen/.local/lib/python3.10/site-packages/pip (python 3.10)
Defaulting to user installation because normal site-packages is not writeable
Processing /home/koen/openvoronoi
  Running command pip subprocess to install build dependencies
  Collecting scikit-build-core@ git+https://github.com/scikit-build/scikit-build-core@henryiii/fix/ninjapath
    Cloning https://github.com/scikit-build/scikit-build-core (to revision henryiii/fix/ninjapath) to /tmp/pip-install-c84tikb2/scikit-build-core_5b4d27c069ce45a897935a5f84545f9e
    Running command git clone --filter=blob:none --quiet https://github.com/scikit-build/scikit-build-core /tmp/pip-install-c84tikb2/scikit-build-core_5b4d27c069ce45a897935a5f84545f9e
    Running command git checkout -b henryiii/fix/ninjapath --track origin/henryiii/fix/ninjapath
    Switched to a new branch 'henryiii/fix/ninjapath'
    Branch 'henryiii/fix/ninjapath' set up to track remote branch 'henryiii/fix/ninjapath' from 'origin'.
    Resolved https://github.com/scikit-build/scikit-build-core to commit 60b6c4894f4880c14aef772d54599aa816f17774
    Running command git submodule update --init --recursive -q
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Collecting exceptiongroup
    Using cached exceptiongroup-1.0.4-py3-none-any.whl (14 kB)
  Collecting packaging>=20.9
    Using cached packaging-22.0-py3-none-any.whl (42 kB)
  Collecting tomli>=1.1
    Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
  Building wheels for collected packages: scikit-build-core
    Building wheel for scikit-build-core (pyproject.toml): started
    Building wheel for scikit-build-core (pyproject.toml): finished with status 'done'
    Created wheel for scikit-build-core: filename=scikit_build_core-0.1.5.dev1+g60b6c48-py3-none-any.whl size=95862 sha256=6f8eb6bb677f106634bea04c2213954a3baed9a936c771e5cfe7889c2d3a6665
    Stored in directory: /tmp/pip-ephem-wheel-cache-5i2fdj7f/wheels/06/66/76/470713fba99dd8a0fff7302ba90f558c3309abab600ad8d159
  Successfully built scikit-build-core
  Installing collected packages: tomli, packaging, exceptiongroup, scikit-build-core
  Successfully installed exceptiongroup-1.0.4 packaging-22.0 scikit-build-core-0.1.5.dev1+g60b6c48 tomli-2.0.1
  Installing build dependencies ... done
  Running command Getting requirements to build wheel
  Getting requirements to build wheel ... done
  Running command pip subprocess to install backend dependencies
  Collecting distlib
    Using cached distlib-0.3.6-py2.py3-none-any.whl (468 kB)
  Collecting ninja>=1.5
    Using cached ninja-1.11.1-py2.py3-none-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (155 kB)
  Collecting pyproject_metadata
    Using cached pyproject_metadata-0.6.1-py3-none-any.whl (7.4 kB)
  Collecting pathspec
    Using cached pathspec-0.10.3-py3-none-any.whl (29 kB)
  Collecting packaging>=19.0
    Using cached packaging-22.0-py3-none-any.whl (42 kB)
  Installing collected packages: ninja, distlib, pathspec, packaging, pyproject_metadata
  Successfully installed distlib-0.3.6 ninja-1.11.1 packaging-22.0 pathspec-0.10.3 pyproject_metadata-0.6.1
  Installing backend dependencies ... done
  Running command Preparing metadata (pyproject.toml)
  2022-12-16 15:34:12,171 - scikit_build_core - DEBUG - RUN: /usr/bin/cmake --version
  2022-12-16 15:34:12,178 - scikit_build_core - INFO - CMake version: 3.22.1
  2022-12-16 15:34:12,186 - scikit_build_core - DEBUG - Default generator: Unix Makefiles
  2022-12-16 15:34:12,187 - scikit_build_core - DEBUG - RUN: /tmp/pip-build-env-bdxraoi5/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja --version
  2022-12-16 15:34:12,189 - scikit_build_core - INFO - Ninja version: 1.11.1
  2022-12-16 15:34:12,189 - scikit_build_core - DEBUG - CMAKE_GENERATOR: Using ninja: /tmp/pip-build-env-bdxraoi5/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja
  2022-12-16 15:34:12,189 - scikit_build_core - DEBUG - /tmp/tmp9wio8igr/build/CMakeInit.txt:
    set(SKBUILD [===[2]===] CACHE STRING "")
    set(SKBUILD_CORE_VERSION [===[0.1.5.dev1+g60b6c48]===] CACHE STRING "")
    set(SKBUILD_PROJECT_NAME [===[openvoronoi]===] CACHE STRING "")
    set(SKBUILD_PROJECT_VERSION [===[2022.12.9]===] CACHE STRING "")
    set(PYTHON_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(PYTHON_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(PYTHON_LIBRARY [===[/usr/lib/aarch64-linux-gnu/libpython3.10.so]===] CACHE PATH "")
    set(Python_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(Python_ROOT_DIR [===[/usr]===] CACHE STRING "")
    set(Python_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(Python_FIND_REGISTRY [===[NEVER]===] CACHE STRING "")
    set(Python3_EXECUTABLE [===[/usr/bin/python3]===] CACHE STRING "")
    set(Python3_ROOT_DIR [===[/usr]===] CACHE STRING "")
    set(Python3_INCLUDE_DIR [===[/usr/include/python3.10]===] CACHE PATH "")
    set(Python3_FIND_REGISTRY [===[NEVER]===] CACHE STRING "")
    set(SKBUILD_SOABI [===[cpython-310-aarch64-linux-gnu]===] CACHE STRING "")
    set(SKBUILD_PLATLIB_DIR [===[/tmp/tmp9wio8igr/wheel/platlib]===] CACHE PATH "")
    set(SKBUILD_DATA_DIR [===[/tmp/tmp9wio8igr/wheel/data]===] CACHE PATH "")
    set(SKBUILD_HEADERS_DIR [===[/tmp/tmp9wio8igr/wheel/headers]===] CACHE PATH "")
    set(SKBUILD_SCRIPTS_DIR [===[/tmp/tmp9wio8igr/wheel/scripts]===] CACHE PATH "")
    set(SKBUILD_NULL_DIR [===[/tmp/tmp9wio8igr/wheel/null]===] CACHE PATH "")
  2022-12-16 15:34:12,189 - scikit_build_core - DEBUG - RUNENV: SHELL=/bin/bash SESSION_MANAGER=local/linuxke:@/tmp/.ICE-unix/910,unix/linuxke:/tmp/.ICE-unix/910 QT_ACCESSIBILITY=1 COLORTERM=truecolor XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg SSH_AGENT_LAUNCHER=gnome-keyring XDG_MENU_PREFIX=gnome- GNOME_DESKTOP_SESSION_ID=this-is-deprecated LC_ADDRESS=de_DE.UTF-8 GNOME_SHELL_SESSION_MODE=ubuntu LC_NAME=de_DE.UTF-8 SSH_AUTH_SOCK=/run/user/1000/keyring/ssh XMODIFIERS=@im=ibus DESKTOP_SESSION=ubuntu LC_MONETARY=de_DE.UTF-8 GTK_MODULES=gail:atk-bridge PWD=/home/koen/openvoronoi LOGNAME=koen XDG_SESSION_DESKTOP=ubuntu XDG_SESSION_TYPE=wayland SYSTEMD_EXEC_PID=930 XAUTHORITY=/run/user/1000/.mutter-Xwaylandauth.29PYW1 HOME=/home/koen USERNAME=koen IM_CONFIG_PHASE=1 LC_PAPER=de_DE.UTF-8 LANG=en_US.UTF-8 LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36: XDG_CURRENT_DESKTOP=ubuntu:GNOME VTE_VERSION=6800 WAYLAND_DISPLAY=wayland-0 GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/f3de64be_dbde_4269_aeab_a91af58fde4c GNOME_SETUP_DISPLAY=:1 LESSCLOSE=/usr/bin/lesspipe %s %s XDG_SESSION_CLASS=user TERM=xterm-256color LC_IDENTIFICATION=de_DE.UTF-8 LESSOPEN=| /usr/bin/lesspipe %s USER=koen GNOME_TERMINAL_SERVICE=:1.101 DISPLAY=:0 SHLVL=1 LC_TELEPHONE=de_DE.UTF-8 QT_IM_MODULE=ibus LC_MEASUREMENT=de_DE.UTF-8 XDG_RUNTIME_DIR=/run/user/1000 LC_TIME=de_DE.UTF-8 XDG_DATA_DIRS=/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop PATH=/tmp/pip-build-env-bdxraoi5/overlay/bin:/tmp/pip-build-env-bdxraoi5/normal/bin:/home/koen/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin GDMSESSION=ubuntu DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus LC_NUMERIC=de_DE.UTF-8 _=/usr/bin/python3 OLDPWD=/home/koen PIP_BUILD_TRACKER=/tmp/pip-build-tracker-thasnimc PYTHONNOUSERSITE=1 PYTHONPATH=/tmp/pip-build-env-bdxraoi5/site PEP517_BUILD_BACKEND=scikit_build_core.build CMAKE_GENERATOR=Ninja CMAKE_MAKE_PROGRAM=/tmp/pip-build-env-bdxraoi5/normal/local/lib/python3.10/dist-packages/ninja/data/bin/ninja
  2022-12-16 15:34:12,189 - scikit_build_core - DEBUG - RUN: /usr/bin/cmake -S. -B/tmp/tmp9wio8igr/build -C/tmp/tmp9wio8igr/build/CMakeInit.txt -DCMAKE_BUILD_TYPE=Release -DBUILD_PYTHON_MODULE=ON -DBoost_ADDITIONAL_VERSIONS=1.80.0;1.79.0;1.78.0;1.77.0;1.76.0;1.75.0;1.74.0;1.73.0;1.72.0;1.71.0;1.70.0 -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-bdxraoi5/overlay/local/lib/python3.10/dist-packages/scikit_build_core/resources/find_python -DCMAKE_PREFIX_PATH:PATH=/usr/local/lib/python3.10/dist-packages;/tmp/pip-build-env-bdxraoi5/overlay/local/lib/python3.10/dist-packages
  loading initial cache file /tmp/tmp9wio8igr/build/CMakeInit.txt
  CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
  CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
  CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
  -- Configuring incomplete, errors occurred!
  See also "/tmp/tmp9wio8igr/build/CMakeFiles/CMakeOutput.log".
  *** scikit-build-core 0.1.5.dev1+g60b6c48 using CMake 3.22.1
  *** Configurating CMake...

  *** CMake configuration failed
  error: subprocess-exited-with-error
  
  \u00d7 Preparing metadata (pyproject.toml) did not run successfully.
  \u2502 exit code: 1
  \u2570\u2500> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/bin/python3 /home/koen/.local/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp3a6pq464
  cwd: /home/koen/openvoronoi
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

\u00d7 Encountered error while generating package metadata.
\u2570\u2500> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

--- a/src/scikit_build_core/_shutil.py
+++ b/src/scikit_build_core/_shutil.py
@@ -43,6 +43,10 @@ class Run:
             os.fspath(arg) if isinstance(arg, os.PathLike) else arg for arg in args
         ]
 
+        if self.env:
+            logger.debug(
+                "RUNENV: {}", " ".join(f"{k}={v}" for k, v in self.env.items())
+            )
         logger.debug("RUN: {}", " ".join(options))
 
         return subprocess.run(

--- a/src/scikit_build_core/builder/builder.py
+++ b/src/scikit_build_core/builder/builder.py
@@ -84,7 +84,10 @@ class Builder:
         if fp_backport and self.config.cmake.version < Version(fp_backport):
             self.config.module_dirs.append(Path(find_python.__file__).parent.resolve())
 
-        set_environment_for_gen(self.config.cmake, self.config.env, self.settings.ninja)
+        local_def = set_environment_for_gen(
+            self.config.cmake, self.config.env, self.settings.ninja
+        )
+        cmake_defines.update(local_def)
 
         cache_config: dict[str, str | Path] = {
             "SKBUILD": "2",

--- a/src/scikit_build_core/builder/generator.py
+++ b/src/scikit_build_core/builder/generator.py
@@ -3,7 +3,7 @@ from __future__ import annotations
 import re
 import subprocess
 import sys
-from collections.abc import MutableMapping
+from collections.abc import Mapping, MutableMapping
 
 from packaging.version import Version
 
@@ -55,7 +55,7 @@ def get_default(cmake: CMake) -> str | None:
 
 def set_environment_for_gen(
     cmake: CMake, env: MutableMapping[str, str], ninja_settings: NinjaSettings
-) -> None:
+) -> Mapping[str, str]:
     """
     This function modifies the environment as needed to safely set a generator.
 
@@ -72,25 +72,28 @@ def set_environment_for_gen(
             # This must also be set when *_PLATFORM is set.
             env.setdefault("CMAKE_GENERATOR", default)
             env.setdefault("CMAKE_GENERATOR_PLATFORM", get_cmake_platform(env))
-    elif env.get("CMAKE_GENERATOR", "Ninja") == "Ninja" and not env.get(
+            return {}
+    if env.get("CMAKE_GENERATOR", "Ninja") == "Ninja" and not env.get(
         "CMAKE_MAKE_PROGRAM", ""
     ):
         ninja = best_program(
             get_ninja_programs(),
             minimum_version=Version(ninja_settings.minimum_version),
         )
-        if ninja is None:
-            msg = "Ninja or make is required to build"
-            if not ninja_settings.make_fallback:
-                raise NinjaNotFoundError(msg)
-
-            make_programs = list(get_make_programs())
-            if not make_programs:
-                raise NinjaNotFoundError(msg)
-            env.setdefault("CMAKE_GENERATOR", "Unix Makefiles")
-            env.setdefault("CMAKE_MAKE_PROGRAM", str(make_programs[0]))
-            logger.debug("CMAKE_GENERATOR: Using make: {}", make_programs[0])
-        else:
+        if ninja is not None:
             env.setdefault("CMAKE_GENERATOR", "Ninja")
-            env.setdefault("CMAKE_MAKE_PROGRAM", str(ninja.path))
             logger.debug("CMAKE_GENERATOR: Using ninja: {}", ninja.path)
+            return {"CMAKE_MAKE_PROGRAM": str(ninja.path)}
+
+        msg = "Ninja or make is required to build"
+        if not ninja_settings.make_fallback:
+            raise NinjaNotFoundError(msg)
+
+        make_programs = list(get_make_programs())
+        if not make_programs:
+            raise NinjaNotFoundError(msg)
+        env.setdefault("CMAKE_GENERATOR", "Unix Makefiles")
+        logger.debug("CMAKE_GENERATOR: Using make: {}", make_programs[0])
+        return {"CMAKE_MAKE_PROGRAM": str(make_programs[0])}
+
+    return {}

